Replicator/Replicator.cc

```diff
@@ -1307,4 +1295,36 @@ namespace litecore { namespace repl {
         }
     }
 
+    void Replicator::setMsgHandlerFor3_0_Client(Retained<blip::MessageIn> request)
+    {
+        if (_setMsgHandlerFor3_0_ClientDone) {
+            return;
+        } else {
+            _setMsgHandlerFor3_0_ClientDone = true;
+        }
+
+        if (_options->isActive()) {
+            return; // only deal with passive replicator.
+        }
+
+        if (request->intProperty(kCollectionProperty, kNotCollectionIndex)
+            != kNotCollectionIndex) {
+            return; // 3.0 message should not include the collection property
+        }
+
+        std::vector<C4CollectionSpec> v;
+        v.emplace_back(kC4DefaultCollectionSpec);
+        _options->rearrangeCollectionsFor3_0_Client();
+        DebugAssert(!_options->collectionAware());
+        DebugAssert(_options->workingCollectionCount() == 1);
+        if (!_options->collectionPath(0)) {
+            logVerbose("Client is legacy 3.0, but the default collection is not in the config of this 3.1 replicator.");
+            // what should the message be? 3.0 client is not aware of default collection.
+            request->respondWithError({"BLIP"_sl, 400, "This server does not support 3.0 client"_sl});
```

Comment:
Suggestion:  This server is not configured for 3.0 client support

Reply:
Agreed, it's better.